### PR TITLE
Adds custom expansions to the listers for apps/v1

### PR DESCRIPTION
**Pull request by kow3ns**
**What this PR does / why we need it**:
We need to add the listers expansions for the apps group version API objects.
```release-note
NONE
```

### CODE REVIEW

**Review by dhilipkumars — APPROVED** [review]
LGTM from my side

**Comment by kow3ns** [comment]
@liggitt can you approve?

File: staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	GetHistoryDaemonSets(history *apps.ControllerRevision) ([]*apps.DaemonSet, error)
+}
+
+// DaemonSetNamespaceListerExpansion allows custom methods to be added to
+// DaemonSetNamespaceLister.
+type DaemonSetNamespaceListerExpansion interface{}
+
+// GetPodDaemonSets returns a list of DaemonSets that potentially match a pod.
+// Only the one specified in the Pod's ControllerRef will actually manage it.
+// Returns an error only if no matching DaemonSets are found.
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go]: I see this is a copy of apps/v1beta2 GetPodDaemonSets, but I'm wondering what the point of returning a daemonset that doesn't match the controllerref is

File: staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	if len(daemonSets) == 0 {
+		return nil, fmt.Errorf("could not find daemon set for pod %s in namespace %s with labels: %v", pod.Name, pod.Namespace, pod.Labels)
+	}
+
+	return daemonSets, nil
+}
+
+// GetHistoryDaemonSets returns a list of DaemonSets that potentially
+// match a ControllerRevision. Only the one specified in the ControllerRevision's ControllerRef
+// will actually manage it.
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go]: same here... what is the benefit of getting back a daemonset that happened to match the controllerrevision's labels, but not the controllerref name?

File: staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go (hunk 1 above)
Comment (kow3ns) [reply staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go]: It's needed for adoption of orphans.

File: staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go (hunk 2 above)
Comment (kow3ns) [reply staging/src/k8s.io/client-go/listers/apps/v1/daemonset_expansion.go]: Same as above.

**Comment by kow3ns** [comment]
@liggit @enisoc @janetkuo do we want to add this for 1.9?

**Comment by liggitt** [comment]
/approve
Makes sense to provide same client helpers as v1beta2

**Comment by liggitt** [comment]
@kubernetes/sig-apps-pr-reviews

**Comment by kow3ns** [comment]
@liggitt needs approve no issue

**Comment by dhilipkumars** [comment]
I think Milestone should be set to `1.9`

**Review by janetkuo — APPROVED** [review]

**Comment by jberkus** [comment]
/priority critical-urgent

/remove-priority important-soon

adjusting priorities for code freeze

**Pull request merged**